[boost] Update to 1.86.0

Osyotr opened the pull request:
Fixes #40558.

Questions:
- Does `boost-cobalt` need platform expression for `boost-context`? Also for `boost-coroutine` (`boost-coroutine2` does not have one).
- How to filter `pfr` from `mysql` more cleanly in `generate-ports.ps1`?
- `boost-exception`: header-only in cmake but static in b2

Osyotr committed:
[boost] Update to 1.86.0

Osyotr committed:
[boost-cmake] Don't honor Boost_USE_STATIC_LIBS

Osyotr committed:
Merge remote-tracking branch 'origin/master' into boost-update

dg0yt commented:
As a drive-by fix: There are some homepage links reported to be broken at https://repology.org/repository/vcpkg/problems.

Osyotr committed:
Fix homepage URL for numeric libraries

Osyotr committed:
[boost-process] Restore opt-filesystem.patch

Osyotr committed:
Merge remote-tracking branch 'origin/master' into boost-update

Osyotr committed:
[pcl] Fix build with Boost 1.86.0

Osyotr committed:
Merge remote-tracking branch 'origin/master' into boost-update

Osyotr committed:
[boost-cmake] Fix handling of optional components

Osyotr committed:
[boost-process] Fix filesystem patch

Osyotr committed:
[boost-process] Disable on android and uwp

It uses 'wordexp.h' which is not available on Android.
It uses parts of WinAPI that are not available on UWP.

Osyotr committed:
[robotraconteur] Add boost-uuid patch

Osyotr committed:
Merge remote-tracking branch 'origin/master' into boost-update

Osyotr committed:
Merge remote-tracking branch 'origin' into boost-update

Osyotr commented:
Currently blocked by compilation errors in the following ports:
`ctbench`
`hazelcast-cpp-client`
`restc-cpp`

yurybura commented:
@Osyotr Is there any progress on this PR?

Osyotr committed:
Merge remote-tracking branch 'origin/master' into boost-update

Osyotr committed:
Add ports broken by boost 1.86 into ci.baseline.txt

Osyotr commented:
> @Osyotr Is there any progress on this PR?

The "Boost" part of this PR is done. Waiting for upstreams to respond.

yurybura commented:
> The "Boost" part of this PR is done. Waiting for upstreams to respond.

@Osyotr, Could you please look at the review comments?

Osyotr commented:
> @Osyotr, Could you please look at the review comments?

I don't see any review comments in this PR.

yurybura commented:
> > @Osyotr, Could you please look at the review comments?
> 
> I don't see any review comments in this PR.

![image](https://github.com/user-attachments/assets/14820dec-12f5-4aba-a433-812cb6c0cf6c)

Osyotr commented:
You have to click "Submit review" first.
![image](https://github.com/user-attachments/assets/bb758f5a-15d0-41d5-ba37-015175c0b080)

yurybura commented:
> You have to click "Submit review" first. 

Oh, sorry. Done.

Osyotr committed:
[boost-locale] Remove stale patch

Osyotr commented on ports/boost-gil/portfile.cmake:
```diff
@@ -4,7 +4,7 @@ vcpkg_from_github(
+    SHA512 97de2f6af8b60f826c330677646ebf9e94832f4e94bbe2be7ec799856956e2f6c809f6bbd00541642e7f6db4817d2f10343ab55babdb6710bb9c1c702c313c6e
     HEAD_REF master
     PATCHES
         remove-boost-filesystem-dep.diff
```
There are no options in vcpkg. Have to force boost-filesystems :(

yurybura replied:
> There are no options in vcpkg. Have to force boost-filesystems :(

I mean the following modification:

```json
  "features": {
    "boost-filesystem": {
      "dependencies": [
        "boost-filesystem"
      ]
    }
  }
```

```cmake
vcpkg_check_features(
  OUT_FEATURE_OPTIONS
    FEATURE_OPTIONS
  FEATURES
    "boost-filesystem" BOOST_GIL_USE_BOOST_FILESYSTEM
)
```

Osyotr replied:
This is not a feature, this is an alternative. Vcpkg does not support alternatives yet.

yurybura replied:
Ok. What about explicitly define BOOST_GIL_USE_BOOST_FILESYSTEM=OFF instead of patching?
The patch removes `boost_filesystem` dependency from the CMake target but this library still may be used in the source code due to BOOST_GIL_USE_BOOST_FILESYSTEM is ON by default.

Osyotr replied:
IIUC none of these options are available to vcpkg because we take the "if" path:
https://github.com/simmplecoder/gil/blob/8994c2ff8ed8b88e0ffef9658a4c603c4a001fad/CMakeLists.txt#L10-L41
The patch is still relevant.

Osyotr committed:
[boost-asio] Remove stale patch

Osyotr commented on ports/boost-odeint/features.cmake:
```diff
@@ -0,0 +1,6 @@
+  OUT_FEATURE_OPTIONS
+    FEATURE_OPTIONS
+  FEATURES
+    "mpi" BOOST_ENABLE_MPI
```
Shouldn't it be a separate feature? And `boost-odeint[mpi]` should depend on that feature

yurybura replied:
> Shouldn't it be a separate feature? And `boost-odeint[mpi]` should depend on that feature

It's up to you. I noticed that the patch is not needed due to feature has been implemented in library.

yurybura commented on ports/boost-build/vcpkg.json:
`boost-build` is not used anymore. This port should be removed.

Osyotr committed:
[boost-build] Delist port

Osyotr committed:
[restc-cpp] Use upstream patch to build with boost 1.86

Osyotr committed:
Merge remote-tracking branch 'origin/master' into boost-update

dg0yt commented:
> boost-build is not used anymore. This port should be removed.

This port contains the b2 executable. How do we know it isn't used by users?

dg0yt commented:
CI regression unrelated to this PR: https://github.com/microsoft/vcpkg-tool/pull/1483#issuecomment-2375896503

yurybura commented:
> > boost-build is not used anymore. This port should be removed.
> 
> This port contains the b2 executable. How do we know it isn't used by users?

I think better to implement a new port for a tool `ports/vcpkg-tool-b2` and scripts like `scripts/cmake/vcpkg_find_acquire_program(B2).cmake` and `scripts/cmake/vcpkg_build_b2.cmake`.

Osyotr committed:
Revert "[boost-build] Delist port"

This reverts commit 4ad2ac2f1e0c4233d61826603d38bbbdd1689c07.

yurybura commented:
@Osyotr Could you please tell how soon we can expect completion of this PR? Thank you in advance.

dg0yt commented:
vcpkg team or PR author could trigger re-run of CI.

Osyotr committed:
Merge remote-tracking branch 'origin/master' into boost-update

Osyotr committed:
[boost-odeint] Add feature adapters

Osyotr committed:
[boost-stacktrace] fix patch

JavierMatosD commented on scripts/boost/generate-ports.ps1:
```diff
@@ -160,14 +127,19 @@ $portData = @{
     };
     "boost-mysql"           = @{ "dependencies" = @("openssl"); };
     "boost-odeint"           = @{
+        "default-features" = @("adapters");
```
Can you explain what this feature does and why it is needed as a default feature?

Osyotr replied:
Reflecting upstream changes:
https://github.com/boostorg/odeint/pull/83

JavierMatosD replied:
The PR description states that the update allows the library to be built with and without. Please see https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#default-features-should-enable-behaviors-not-apis

This means that it is very rare for a feature to belong to default-features.

Osyotr replied:
Yes and `BOOST_NUMERIC_ODEINT_NO_ADAPTORS` is not set by default => `boost-odeint` uses adapters by default.

Osyotr replied:
Note: I'm not a `boost-odeint` user and I'm fine with reverting this change or moving it into a separate PR.

Osyotr committed:
Revert "[boost-odeint] Add feature adapters"

This reverts commit 8a7f9800d63bcbe9c439af50a0dbb82f4ef2796e.

JavierMatosD commented on scripts/ci.baseline.txt:
```diff
@@ -446,6 +447,18 @@ gz-tools2:x64-android=fail
 gz-gui7:x64-osx=fail
 halide:x64-windows-static=fail
+# hazelcast-cpp-client fails to build with Boost 1.86
+hazelcast-cpp-client:x86-windows=fail
+hazelcast-cpp-client:x64-windows=fail
+hazelcast-cpp-client:x64-windows-static=fail
+hazelcast-cpp-client:x64-windows-static-md=fail
+hazelcast-cpp-client:arm64-windows=fail
+hazelcast-cpp-client:x64-osx=fail
+hazelcast-cpp-client:arm64-osx=fail
+hazelcast-cpp-client:x64-linux=fail
+hazelcast-cpp-client:arm-neon-android=fail
+hazelcast-cpp-client:arm64-android=fail
+hazelcast-cpp-client:x64-android=fail
```
We shouldn't just be marking ports as =fail if this update breaks them. They should be fixed accordingly..

Osyotr replied:
The upstreams are notified but no response:
https://github.com/hazelcast/hazelcast-cpp-client/issues/1245
https://github.com/JPenuchot/ctbench/issues/39

BillyONeal replied:
It's surprising to me that ctbench would be working at all, since it claims to need gcc12, but the test lab has gcc11. Is that actually a regression?

BillyONeal replied:
Hmmm it did pass in https://dev.azure.com/vcpkg/public/_build/results?buildId=107918&view=logs&j=f79cfdd7-47a8-597f-8f57-dc3e21a8f2ad&t=4f03addf-ef5a-50a2-71be-19894d9df4e3

BillyONeal replied:
I spent quite a while trying to fix ctbench; it isn't clear to me exactly what's going on. There's some games happening with conversions to and from std::string :(.

BillyONeal replied:
@ras0219-msft pointed out upstream has a commit that should fix, it worked in my testing: https://github.com/Osyotr/vcpkg/pull/184

IYanakiev34 commented:
When is this pull request expected to be merged?

BillyONeal committed:
[hazelcast-cpp-client] Fix build with Boost 1.86.

BillyONeal committed:
Use the shorter patch

BillyONeal committed:
version db

BillyONeal committed:
[ctbench] Apply upstream fix for Boost 1.86.

BillyONeal commented:
@Osyotr Can you take a look at https://github.com/Osyotr/vcpkg/pull/183 and https://github.com/Osyotr/vcpkg/pull/184 ?

Osyotr committed:
Merge pull request #183 from BillyONeal/boost-186-hazelcast

[hazelcast-cpp-client] Fix build with Boost 1.86.

Osyotr committed:
Merge pull request #184 from BillyONeal/boost-update-ctbench

[ctbench] Apply upstream fix for Boost 1.86.
